crates/interpreter/src/valid.rs

```diff
@@ -885,7 +888,12 @@ impl<'a, 'm> Expr<'a, 'm> {
         let n = self.labels.len();
         check(l < n)?;
         let source = self.branch_source();
+        // Apply the side table entry in flash and compute the expected target branch.
+        // source.parser += delta_ip;
+        // source.branch_table += delta_stp;
         let label = &mut self.labels[n - l - 1];
+        // if label.branch.is_none() { label.branch = source; }
+        // else { compare label.branch.unwrap() with source to make sure they are the same }
```

Comment:
Yes on the idea, but the Rust code should use a match, or something like `label.branch.replace(source).is_none_or(|x| x == source)`.